sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -130,11 +135,49 @@ export class GlobalEndpointManager {
     return canUse;
   }
 
+  private getEffectiveExcludedLocations(
+    options: RequestOptions | FeedOptions | undefined,
+    resourceType: ResourceType,
+  ): Set<string> {
+    if (!canApplyExcludedLocations(resourceType)) {
+      return new Set<string>();
+    }
+
+    if (
+      !options ||
+      !Object.prototype.hasOwnProperty.call(options, "excludedLocations") ||
+      options.excludedLocations === undefined ||
+      options.excludedLocations === null
+    ) {
+      return this.clientLevelExcludedLocations;
+    }
+
+    if (options.excludedLocations.length === 0) {
+      return new Set<string>();
+    }
+
+    return new Set(options.excludedLocations.map((loc) => normalizeEndpoint(loc)));
+  }
+
+  private filterExcludedLocations(
+    preferredLocations: string[],
+    excludedLocations?: Set<string>,
+  ): string[] {
+    if (!excludedLocations || excludedLocations.size === 0) {
+      return preferredLocations;
+    }
+    const filteredLocations = preferredLocations.filter(
+      (location) => !excludedLocations?.has(normalizeEndpoint(location)),
+    );
+    return filteredLocations.length > 0 ? filteredLocations : preferredLocations;
+  }
+
   public async resolveServiceEndpoint(
     diagnosticNode: DiagnosticNodeInternal,
     resourceType: ResourceType,
     operationType: OperationType,
     startServiceEndpointIndex: number = 0, // Represents the starting index for selecting servers.
+    requestOptions?: RequestOptions | FeedOptions | undefined, // add to support request-level excluded region(location) overrides
```

Comment:
can we make it mandatory? 
We can create a default (empty) request option when user is not pasing it in request.